circle.yml

```diff
@@ -1,35 +1,84 @@
+defaults: &defaults
+  working_directory: ~/repo
+  docker:
+    - image: applicaster/zapp:0.8
+
+whitelist: &whitelist
+  paths:
+    - node_modules/*
+    - src/*
+    - LICENSE
+    - package.json
+    - package-lock.json
+    - README.md
+    - .eslintrc
+    - .babelrc
+
 version: 2
 jobs:
   build:
-    docker:
-      - image: applicaster/zapp:0.8
+    <<: *defaults
     steps:
       - restore_cache:
           key: react-native-zapp-bridge-{{ .Branch }}-{{ .Revision }}
           keys:
             - react-native-zapp-bridge-{{ .Branch }}-{{ .Revision }}
             - react-native-zapp-bridge-{{ .Branch }}-
             - react-native-zapp-bridge-
+
       - checkout
+
       - save_cache:
           key: react-native-zapp-bridge-{{ .Branch }}-{{ .Revision }}
           paths:
             - ".git"
+
       - restore_cache:
           keys:
             - npm-{{ checksum "package-lock.json" }}
             - npm
+
       - run:
           name: Install javascript dependencies
           command: npm i
+
       - save_cache:
           key: npm-{{ checksum "package-lock.json" }}
           paths:
             - "~/.npm"
             - node_modules
+
+      - persist_to_workspace:
+          root: ~/repo
+          <<: *whitelist
+
       - run:
           name: Linting - javascript
           command: npm run lint
+
       - run:
           name: Javascript tests
           command: npm t
+
+  deploy:
+    <<: *defaults
+
+    steps:
+      - attach_workspace:
+          at: ~/repo
+
+      - run:
+          name: Semantic release
+          command: npx semantic-release --dry-run
```

Comment:
for the dry run its ok, but you will probably want to publish only when pushing a specific branch

Reply:
I agree, but we should define a way to publish as release candidate to - maybe not each branch, but it could be useful

Reply:
branches are filtered below so only changes to master will result in publish jobs

Reply:
Sounds good, please add it to readme as well :)